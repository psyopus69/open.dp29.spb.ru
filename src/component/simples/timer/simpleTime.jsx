import React, {Fragment} from "react";
import {PropTypes} from "prop-types";

// Simple for HOC withTimer
const SimpleTime = (props) => {
  const {renderTimer, tickTimer, setTime} = props;
  return <Fragment>
    <div>Test 1.2 SimpleTime React</div>
    <div>Timer: {renderTimer()}</div>
    <div>Controls Tick: <button onClick={() => {
      tickTimer();
    }}>Tick</button></div>
    <div>Set time: <input name='setTime' onChange={(event) => {
      setTime(parseInt(event.target.value, 10));
    }}
    /></div>
  </Fragment>;
};
SimpleTime.propTypes = {
  renderTimer: PropTypes.func.isRequired,
  tickTimer: PropTypes.func.isRequired,
  setTime: PropTypes.func.isRequired,
};

export {
  SimpleTime,
};
