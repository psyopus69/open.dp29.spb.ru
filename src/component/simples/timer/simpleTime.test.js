import React from "react";
import renderer from "react-test-renderer";
import {SimpleTime} from "./simpleTime";

it(`panel with time & setTimeAction & tickTime`, () => {
  const tree = renderer.create(<SimpleTime renderTimer={jest.fn()} setTime={jest.fn()} tickTimer={jest.fn()}
  />);
  expect(tree).toMatchSnapshot();
});
