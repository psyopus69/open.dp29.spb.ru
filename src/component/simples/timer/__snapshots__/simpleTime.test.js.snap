// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`panel with time & setTimeAction & tickTime 1`] = `
Array [
  <div>
    Test 1.2 SimpleTime React
  </div>,
  <div>
    Timer: 
  </div>,
  <div>
    Controls Tick: 
    <button
      onClick={[Function]}
    >
      Tick
    </button>
  </div>,
  <div>
    Set time: 
    <input
      name="setTime"
      onChange={[Function]}
    />
  </div>,
]
`;
